infoschema/tables.go

```diff
@@ -1997,7 +2011,81 @@ func dataForTiDBClusterInfo(ctx sessionctx.Context) ([][]types.Datum, error) {
 	return rows, nil
 }
 
-func dataForClusterConfig(ctx sessionctx.Context) ([][]types.Datum, error) {
+func dataForClusterLoadInfo(ctx sessionctx.Context) ([][]types.Datum, error) {
+	serversInfo, err := getClusterServerInfo(ctx)
+	if err != nil {
+		return nil, err
+	}
+	ipMap := make(map[string]struct{}, len(serversInfo))
+	rows := make([][]types.Datum, 0, len(serversInfo)*10)
+	for _, srv := range serversInfo {
+		// TODO: remove this after PD/TiKV support diagnostic grpc service.
+		if srv.tp == "pd" || srv.tp == "tikv" {
+			continue
+		}
+		addr := srv.statusAddr
+		if _, ok := ipMap[addr]; ok {
+			continue
+		}
+		ipMap[addr] = struct{}{}
+
+		items, err := getServerInfoByGRPC(srv.statusAddr, diagnosticspb.ServerInfoType_LoadInfo)
+		if err != nil {
+			return nil, err
+		}
+		partRows := serverInfoItemToRows(items, srv.tp, srv.statusAddr)
+		rows = append(rows, partRows...)
+	}
+	return rows, nil
+}
+
+func serverInfoItemToRows(items []*diagnosticspb.ServerInfoItem, tp, addr string) [][]types.Datum {
+	rows := make([][]types.Datum, 0, len(items))
+	for _, v := range items {
+		for _, item := range v.Pairs {
+			row := types.MakeDatums(
+				tp,
+				addr,
+				v.Tp,
+				v.Name,
+				item.Key,
+				item.Value,
+			)
+			rows = append(rows, row)
+		}
+	}
+	return rows
+}
+
+func getServerInfoByGRPC(address string, tp diagnosticspb.ServerInfoType) ([]*diagnosticspb.ServerInfoItem, error) {
+	conn, err := grpc.Dial(address, grpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := conn.Close()
+		if err != nil {
+			log.Error("close grpc connection error", zap.Error(err))
+		}
+	}()
+
+	cli := diagnosticspb.NewDiagnosticsClient(conn)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
It's better to share session context here to avoid leaving goroutine over after killing the connection.